src/goto-programs/goto_convert.cpp

```diff
@@ -344,14 +344,49 @@ void goto_convertt::convert_label(
 
   goto_programt tmp;
 
-  // magic thread creation label?
+  // magic thread creation label, see ansi-c/library/pthread_lib.c
   if(has_prefix(id2string(label), "__CPROVER_ASYNC_"))
   {
-    // this is like a START_THREAD
-    codet tmp_code(ID_start_thread);
-    tmp_code.copy_to_operands(code.op0());
-    tmp_code.add_source_location()=code.source_location();
-    convert(tmp_code, tmp);
+    // A: START_THREAD : C
+    // B: GOTO Z
+    // C  SKIP
+    // D: BLOCK
+    // E: END_THREAD
+    // Z: SKIP
+
+    thread_started_counter+=1;
+    const std::string &lbl1 = "TS_" +
```

Comment:
Could these variable names be given more informative names (what does `lbl` stand for?)

Reply:
lbl = label, these variables hold names of the label that  the goto instruction is targeting.  will name them as following instead label_name_c/z